crates/node-core/src/args/debug.rs

```diff
@@ -28,6 +28,38 @@ pub struct DebugArgs {
     #[arg(long = "debug.max-block", help_heading = "Debug")]
     pub max_block: Option<u64>,
 
+    /// Runs a fake consensus client that advances the chain using recent block hashes
+    /// on Etherscan. If specified, requires an `ETHERSCAN_API_KEY` environment
+    #[arg(
+        long = "debug.etherscan",
+        help_heading = "Debug",
+        conflicts_with = "tip",
+        conflicts_with = "rpc_consensus_http"
+    )]
+    pub etherscan: bool,
+
+    /// Runs a fake consensus client using blocks from an RPC endpoint. HTTP endpoint for getting
+    /// past blocks.
+    #[arg(
+        long = "debug.rpc-consensus-http",
+        help_heading = "Debug",
+        conflicts_with = "tip",
+        conflicts_with = "etherscan",
+        requires = "rpc_consensus_ws"
+    )]
+    pub rpc_consensus_http: Option<String>,
+
+    /// Runs a fake consensus client using blocks from an RPC endpoint. WS endpoint for getting new
+    /// blocks.
+    #[arg(
+        long = "debug.rpc-consensus-ws",
+        help_heading = "Debug",
+        conflicts_with = "tip",
+        conflicts_with = "etherscan",
+        requires = "rpc_consensus_http"
+    )]
+    pub rpc_consensus_ws: Option<String>,
```

Comment:
technically we don't need both endpoints, but I'm fine with this limitation for now, but a single arg is sufficient

Reply:
forgot that you can WS works for all calls, not just block streaming 🤦 now it requires just a WS endpoint. I think a HTTP RPC mode might be a follow up PR (where I'd periodically query HTTP API like I do with Etherscan).